### PR TITLE
fix(tables): apply table columns sizes when `editable=false`

### LINKED ISSUE #1937: Resized table columns is not reflected when editable is set to false
### Summary

We're using the Remirror editor to edit and display the content. After resizing the table columns and setting the editor mode to false, the table columns are back to equal width instead of the resized width.

### Steps to reproduce

Create an editor, resize the column, save the changes and display the content when the editor has the `editable` prop set to false.

### Expected results

Columns width should show the width set by the user

### Actual results

Columns width reverts back to equal width

### Possible Solution

N.A

### Screenshot(s)

Table issue

### PR DESCRIPTION
**Pull request by whawker**
### Description

Fixes #1937

Also adds a helper to the doc extension for determining if the doc content has been changed from the default

### Checklist

- [x] I have read the [**contributing**](https://github.com/remirror/remirror/blob/HEAD/docs/contributing.md) document.
- [x] My code follows the code style of this project and `pnpm fix` completed successfully.
- [x] I have updated the documentation where necessary.
- [x] New code is unit tested and all current tests pass when running `pnpm test`.

### CODE REVIEW

File: packages/remirror__extension-tables/src/table-extensions.ts [hunk 1]
```diff
@@ -38,9 +40,14 @@ import {
   setCellAttr,
   splitCell,
   tableEditing,
+  TableMap,
+  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+  // @ts-ignore it is exported
```
Comment (ocavue) [inline packages/remirror__extension-tables/src/table-extensions.ts:45]: let's use `// @ts-expect-error`

File: packages/remirror__extension-tables/src/table-extensions.ts [hunk 2]
```diff
@@ -144,6 +149,43 @@ export class TableExtension extends NodeExtension<TableOptions> {
     return plugins;
   }
 
+  createPlugin(): CreateExtensionPlugin {
```
Comment (ocavue) [inline packages/remirror__extension-tables/src/table-extensions.ts:152]: Maybe we can directly use `createNodeViews` method here. The code will be simpler.

**Review by ocavue — APPROVED** [review]

File: packages/remirror__extension-tables/src/table-extensions.ts (hunk 2 above)
Comment (whawker) [reply packages/remirror__extension-tables/src/table-extensions.ts:152]: We're using `updateExtensionPlugins` here (in the `onView` method),

This only checks the extension's `createPlugin` and `createExternalPlugins` methods.

[See here](https://github.com/remirror/remirror/blob/main/packages/remirror__core/src/builtins/plugins-extension.ts#L193-L216)

**Pull request merged**